bundler/lib/bundler/settings.rb

```diff
@@ -595,5 +596,30 @@ def self.key_to_s(key)
         end
       end
     end
+
+    def credentials_from_helper(uri)
+      helper_key = "credential.helper.#{uri.host}"
+      helper_path = self[helper_key]
+      return unless helper_path
+
+      begin
+        require "shellwords"
+        command = Shellwords.shellsplit(helper_path)
+        command[0] = if command[0].start_with?("/", "~")
+          command[0]
+        else
+          "bundler-credential-#{command[0]}"
+        end
+
+        output = Bundler.with_unbundled_env { IO.popen(command, &:read) }
+        output&.strip unless output.to_s.empty?
```

Comment:
This line is a bit hard for me to read. Are you intentionally returning nil if output is empty or nil? 

If so, can we do that explicitly so that intention is clear?

```suggestion
        output = output.to_s.strip
        output.empty? ? nil : output
```

Reply:
Yes, This one looks better.